documentation/sphinx/source/tss.rst

```diff
@@ -0,0 +1,115 @@
+
+Architecture Overview
+=====================
+
+TSS Mapping
+-----------
+
+The *TSS Mapping*, the mapping containing the (SS Storage ID, TSS Storage ID) pairs for all active TSS pairs in the cluster, is stored in the system keyspace under ``\xff/tss/``.
+
+This information is lazily propagated to clients in the ``KeyServerLocationsRequest`` from the Commit Proxies, which read the TSS mapping from the Transaction State Store.
+
+Clients will use this mapping to duplicate any request to a storage server that has a TSS pair.
+
+Other cluster components that need the full, up-to-date TSS mapping (MoveKeys, ConsistencyCheck, etc...) read it directly from the system keyspace as part of a transaction.
+
+Recruitment
+-----------
+
+TSS Pair recruitment is designed to use the *Perpetual Storage Wiggle*, but will also work if multiple storage processes are added to the cluster.
+
+When Data Distribution detects a deficit of TSS processes in the cluster, it will begin recruiting a TSS pair.
+The pair recruitment logic is as follows:
+
+* Once DD gets a candidate worker from the Cluster Controller, hold that worker as a desired TSS process.
+* Once DD gets a second candidate worker from the Cluster Controller, initialize that worker as a normal SS.
+* Once the second candidate worker is successfully initialized, initialize the first candidate worker as a TSS, passing it the storage ID, starting tag + version, and other information from its SS pair. Because the TSS reads from the same tag starting at the same version, it is guaranteed to recieve the same mutations and data movements as its pair.
+
+One implication of this is, during TSS recruitment, the cluster is effectively down one storage process until a second storage process becomes available.
+While clusters should be able to handle being down a single storage process anyway to tolerate machine failure, an active TSS recruitment will be cancelled if the lack of that single storage process is causing the cluster to be unhealthy. Similarly, if the cluster is unhealthy and unable to find new teams to replicate data to, any existing TSS processes may be killed to make room for new storage servers.
+
+TSS Process
+-----------
+
+A TSS process is almost identical to a normal Storage process, with the exception that it does not pop its tag from the TLog mutation stream, and it listens to and processes private mutations for both itself and its SS pair.
+
+
+Caveats
+=======
+
+Despite its usefulness, the TSS feature does not give a perfect performance comparison of the storage engine(s) under test.
+
+Because it is only enabled on a small percentage of the cluster and only compares single storage processes for the same workload, it may miss potential aggregate performance problems, such as the testing storage engine overall consuming more cpu/memory. It is not a substitute for full-cluster performance and correctness testing or simulation testing.
```

Comment:
```suggestion
Because it is only enabled on a small percentage of the cluster and only compares single storage processes for the same workload, it may miss potential aggregate performance problems, such as the testing storage engine overall consuming more cpu/memory, especially on a host with many instances of it. 

TSS testing using the recommended small number of TSS pairs may also miss performance pathologies from workloads not experienced by the specific storage teams with TSS pairs in their membership.  

TSS testing is not a substitute for full-cluster performance and correctness testing or simulation testing.
```